### ISSUE TITLE
Extend `nvim_list_bufs` to include option 

### ISSUE THREAD

**Issue by ryuheechul** [neovim/neovim#21452]
### Feature already in Vim?

_I don't think so_

### Feature description

`vim.api.nvim_list_bufs()` currently returns buffer numbers for all including unlisted.

Which limits usage code to have to unlisted ones like this -
https://github.com/akinsho/bufferline.nvim/blob/4ecfa81e470a589e74adcde3d5bb1727dd407363/lua/bufferline/utils/init.lua#L132-L143.

And unfortunately when bufferlin.nvim have to deal with lots of unlisted buffers

it slows down significantly even when just switching buffers.

in my case it was ~1577 (mostly unlisted) buffers and it took around `416ms` together and each took `264us`. It could have been just `~2ms` instead (in case with 10 listed buffers) if `nvim_list_bufs` was capable with returning only listed buffers.

<img width="1486" alt="Screenshot 2022-12-16 at 21 27 27" src="https://user-images.githubusercontent.com/2637709/208226876-c86a72a4-bb21-45a5-98e8-5a1fd0fba7b2.png">

And lots of unlisted buffers can happen as easily as simply using [pyright lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#pyright) which load all the python files in the project and put it into unlisted buffer.

given observing `:ls!` and `:ls` work instantaneously, I believe Neovim is capable of returning just listed buffer numbers quickly and hence hope this becomes a feature with an option.

A pseudo API would be like below.

```lua
vim.api.nvim_list_bufs { listed_only = true }
```

Then this way, existing code should work as expected and users will be given an option to receive only listed buffer numbers which could be useful to avoid having to filter it by themselves especially when it hits performance hard.

This addition should make sense as we have `:ls` not only `:ls!`.

**Comment by clason** [comment]
We can't do this for `nvim_xxx` API functions (`:h api-contract`); we'd have to add a new function that takes an `opt` table. _If_ we were to add one, we should do this consistently for all `nvim_list_foo` functions.

**Comment by zeertzjq** [comment]
And
```
Do NOT use these deprecated verbs:
    - list      Redundant with "get"
```

**Comment by clason** [comment]
Yes, that's another reason to deprecate them.

**Comment by ryuheechul** [comment]
I see and thanks for the comments and I just read `:h api-contract`. 
I think the essence of this issue is more for enabling the functionality, so I wouldn't care too much about which API signature will do the job but according to `:h api-contract` function signature can't change so `nvim_list_bufs` seems to be not able to the job as suggested from other comments here.

**Comment by ryuheechul** [comment]
I actually just managed to resolve the lag regarding this issue.

However the case that I experienced only happen when there are multiple different conditions meet at the same time which is explained at https://github.com/ryuheechul/dotfiles/commit/d46c8d75b4e0a60b124476f00cf6e39dbc017bc5#diff-08aceafafc898ec4baf03c0384e7d3d1c0ff0fafba0a10b6391cbc90d4a1d136R13 (it's also my resolution).

Since that no one actually filed similar issue before and I'm happy with my current workaround, I will close this issue so that this issue doesn't distract the focus for other priorities.
